recipe-server/client/control/routerUtils.js

```diff
@@ -0,0 +1,43 @@
+export const searchRouteTree = (tree, name, currentUrl = '') => {
+  // If we have the slug, we have the complete route.
+  if (tree.slug === name) {
+    return currentUrl;
+  }
+
+  // If the slug doesn't match, iterate over the given route tree until
+  // we find one that matches what we need (if any exist).
+  const keys = Object.keys(tree);
+  for (let i = 0; i < keys.length; i += 1) {
+    const curPath = keys[i];
```

Comment:
Nit: I think this could be done as `for (const curPath of Object.keys(tree)) {`,